pkg/registry/apis/provisioning/files.go

```diff
@@ -96,11 +96,17 @@ func (s *filesConnector) Connect(ctx context.Context, name string, opts runtime.
 			return
 		}
 
-		var obj *provisioning.ResourceWrapper
-		ref := r.URL.Query().Get("ref")
-		message := r.URL.Query().Get("message")
-		logger = logger.With("ref", ref, "message", message)
+		query := r.URL.Query()
+		ref := query.Get("ref")
+		if ref != "" {
+			logger = logger.With("ref", ref)
+		}
+		message := query.Get("message")
+		if message != "" {
+			logger = logger.With("message", message)
+		}
```

Comment:
I'd always include these values in the logger, tbh. If it's empty, it usually has a reason to be, and it makes it painfully obvious in the logs that they are in fact empty.